src/groovy/org/pih/warehouse/importer/CSVUtils.groovy

```diff
@@ -153,6 +153,12 @@ class CSVUtils {
         return new CSVPrinter(new StringBuilder(), CSVFormat.DEFAULT)
     }
 
+    static String findSeparator(String firstLine) {
+        List<String> allowedSeparators = [Constants.CUSTOM_COLUMN_SEPARATOR, Constants.TAB_COLUMN_SEPARATOR]
+        String separator = allowedSeparators.find{ String it -> firstLine.contains(it) }
```

Comment:
If a file will be tab-separated and in the first column you will have a semicolon, then this won't work.

Reply:
I know it won't work, but since the previous logic also didn't handle the case, where you'd have a semicolon in the first line and the separator was meant to be comma, it wouldn't work, so I thought we agreed then, that we didn't want to handle such an edge case.
But if you want to me now to handle this edge case, I'll do my best to come up with something

Reply:
In that case you have to execute the whole algorithm for each column separator, not just check if the character is in the first line.
The reason why there was a `firstLine.contains(Constants.CUSTOM_COLUMN_SEPARATOR)` was like a first step:
- if there is `;` character in the first line there is  a chance that it might be a column separator
- then try to parse this first line with this `;` character 
- if the amount of parsed columns equals the expected column count then there is a high probability that the given character `;` is a column separator.
- if not then fallback on the default column separator which is a comma. 

If we want to handle many different characters as potential column separators, then we need to check all of these steps above for each of the characters in a potential character list
_pseudo code:_
```
List<String> allowedSeparators = [";", "\t"]
for (def separator in allowedSeparators) {
  if (firstLine.contains(separator)) { 
    def csvMapReader = new CSVMapReader(new StringReader(firstLine), [separatorChar: separator])

    if (csvMapReader.initFieldKeys().size() == columnCount) {
      return separator
      }
  }
}
// if none of them passed the test then fallback on comma
return ","
```

Reply:
@kchelstowski :point_up: